tests/unit/worker/test_submission_worker.py

```diff
@@ -109,3 +113,43 @@ def test_get_or_create_sqs_queue_for_non_existing_queue(self):
         queue_url = self.sqs_client.get_queue_url(QueueName='test_queue_2')['QueueUrl']
         self.assertTrue(queue_url)
         self.sqs_client.delete_queue(QueueUrl=queue_url)
+
+
+class TestDownloadAndExtractFile(APITestCase):
+    def setUp(self):
+        super(TestDownloadAndExtractFile, self).setUp()
+
+        self.base_dir = "{}/JdyW0ZCFLr".format(tempfile.gettempdir())
+        self.download_filename = "VwYOaVWkdw"
+        self.download_path = "{}/{}".format(self.base_dir, self.download_filename)
+        self.extract_dirname = "gMTblaJFbM"
+        self.extract_path = "{}/{}".format(self.base_dir, self.extract_dirname)
+        self.correct_md5 = "86629294bdc8256500ca37a9afecbd79"
```

Comment:
@grave-panda can we change this to be computed from the file being used in the test. Hardcoding it will be a problem if we change the zip file later on

Reply:
Well I just download a random zip file from the internet(I googled sample zip file download and went with the first link: https://file-examples.com/index.php/text-files-and-archives-download/). So, unless you know about a CDN, or can put a sample zip file somewhere in this project, I cannot change this. Also, if I understand this correctly, testing is checking the code against arbitrary inputs, so this is the expected output. In this case, if I do not hardcore the hash, some bug in future versions of hashlib might cause this test to behave incorrectly.

Reply:
Well I just download a random zip file from the internet(I googled sample zip file download and went with the first link: https://file-examples.com/index.php/text-files-and-archives-download/). So, unless you know about a CDN, or can put a sample zip file somewhere in this project, I cannot change this. Also, if I understand this correctly, testing is checking the code against arbitrary inputs, so this is the expected output. In this case, if I do not hardcore the hash, some bug in future versions of hashlib might cause this test to behave incorrectly.